presto-tests/src/test/java/io/prestosql/tests/TestInformationSchemaConnector.java

```diff
@@ -175,6 +183,20 @@ public void testMetadataCalls()
                 "VALUES 0",
                 new MetadataCallsCount()
                         .withListTablesCount(1));
+        assertMetadataCalls(
+                "SELECT count(*) FROM (SELECT * from test_catalog.information_schema.columns LIMIT 1)",
+                "VALUES 1",
+                new MetadataCallsCount()
+                        .withListSchemasCount(1)
+                        .withListTablesCount(0)
+                        .withGetColumnsCount(8));
+        assertMetadataCalls(
+                "SELECT count(*) FROM (SELECT * from test_catalog.information_schema.columns LIMIT 10000)",
+                "VALUES 10000",
+                new MetadataCallsCount()
+                        .withListSchemasCount(1)
+                        .withListTablesCount(1)
+                        .withGetColumnsCount(100008));
```

Comment:
can you add these test in one commit before so we could see how much loading data lazily helped here?